src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6276,247 +6275,207 @@ private TypeSyntax ParsePointerTypeMods(TypeSyntax type)
         public StatementSyntax ParseStatement()
         {
             return ParseWithStackGuard(
-                () => TryParseStatementCore() ?? ParseExpressionStatement(attributes: default),
+                () => ParseStatementCore(isGlobalScriptLevel: false) ?? ParseExpressionStatement(attributes: default),
```

Comment:
I arbitrarily let the name change back to ParseStatementCore, but it can still return null, even if isGlobalScriptLevel is false.